files/zh-cn/learn/tools_and_testing/client-side_javascript_frameworks/svelte_typescript/index.md

```diff
@@ -0,0 +1,997 @@
+- 更安全的重构：类型使集成开发环境更了解你的代码，并在重构大量代码时为你提供帮助。
+- 类型推断：即使不声明变量类型，也能利用许多 TypeScript 的特性。
+- 支持新的和未来的 JavaScript 特性：TypeScript 将许多最新的 JavaScript 特性转译为普通的老式 JavaScript，使你能够在尚未本地支持这些特性的用户代理上使用它们。
+
+TypeScript 也有一些缺点：
+
+- 非真正的静态类型：类型仅在编译时检查，并且在生成的代码中被删除。
+- 学习曲线陡峭：尽管 TypeScript 是 JavaScript 的超集而不是全新的语言，但学习曲线相当陡峭，尤其是在你没有任何静态语言（如 Java 或 C＃）的经验的情况下。
+- 更多的代码：你需要编写和维护更多的代码。
+- 不能替代自动化测试：尽管类型可以帮助你捕捉到几种错误，但 TypeScript 不能真正替代全面的自动化测试套件。
+- 冗余代码：使用类型、类、接口和泛型可能导致过度设计的代码库。
+
+似乎有广泛的共识认为，TypeScript 特别适用于大型项目，比如许多开发人员在同一个代码库上工作的情况下。事实上，它确实被一些大型项目使用，如 Angular 2、Vue 3、Ionic、Visual Studio Code、Jest，甚至 Svelte 编译器。尽管如此，一些开发人员甚至喜欢在像我们正在开发的小型项目中使用它。
+
+最终，这取决于你的决定。在接下来的几节中，我们希望为你提供更多信息，以便你做出决策。
+
+## 从头创建 Svelte TypeScript 项目
+
+你可以使用[标准模板](https://github.com/sveltejs/template)开始新的 Svelte TypeScript 项目。你只需要运行以下终端命令（在你存储 Svelte 测试项目的位置运行它们——它会创建一个新的目录）：
+
+```bash
+npx degit sveltejs/template svelte-typescript-app
+
+cd svelte-typescript-app
+
+node scripts/setupTypeScript.js
+```
+
+这将创建包含 TypeScript 支持的起始项目，然后你可以根据需要进行修改。
+
+然后，你需要让 npm 下载依赖并以开发模式启动项目，就像我们通常做的那样：
+
+```bash
+npm install
+
+npm run dev
+```
+
+## 向现有的 Svelte 项目添加 TypeScript 支持
+
+要向现有的 Svelte 项目添加 TypeScript 支持，你可以[按照这些说明](https://svelte.dev/blog/svelte-and-typescript#Adding_TypeScript_to_an_existing_project)进行操作。或者，你可以将 [`setupTypeScript.js`](https://github.com/sveltejs/template/blob/master/scripts/setupTypeScript.js) 文件下载到项目根文件夹内的 `scripts` 文件夹中，然后运行 `node scripts/setupTypeScript.js`。
+
+你甚至可以使用 `degit` 下载该脚本。这就是我们将要做的操作，开始将应用程序转换为 TypeScript。
+
+> **备注：** 在你开始将其转换为 TypeScript 之前，你可以运行 `npx degit opensas/mdn-svelte-tutorial/07-typescript-support svelte-todo-typescript` 来获取完整的 JavaScript 写的待办事项列表应用程序。
+
+进入项目的根目录并输入以下命令：
+
+```bash
+npx degit sveltejs/template/scripts scripts       # 将脚本文件下载到 scripts 文件夹
+
+node scripts/setupTypeScript.js                   # 运行它
+# 转换为 TypeScript。
+```
+
+你需要重新运行依赖管理器以开始使用。
+
+```bash
+npm install                                       # 下载新的依赖项
+
+npm run dev                                       # 以开发模式启动应用程序
+```
+
+这些指令适用于任何你想要转换为 TypeScript 的 Svelte 项目。只需注意，Svelte 社区正在不断改进 Svelte 对 TypeScript 的支持，因此你应定期运行 `npm update` 来获取最新的更改。
+
+> **备注：** 如果你在 Svelte 应用程序中使用 TypeScript 时遇到任何问题，请查看[有关 TypeScript 支持的故障排除/常见问题解答部分](https://github.com/sveltejs/language-tools/blob/master/docs/preprocessors/typescript.md#troubleshooting--faq)。
+
+我们之前提到过，TypeScript 是 JavaScript 的超集，因此你的应用程序无需修改即可运行。目前，该应用程序仅是启用了 TypeScript 支持的常规 JavaScript 应用程序，而不会利用 TypeScript 提供的任何特性。你现在可以逐步添加类型。
+
+一旦你配置好了 TypeScript，你就可以通过在组件的脚本部分开头添加 `<script lang='ts'>` 来在 Svelte 组件中使用它。要通过常规的 JavaScript 文件使用它，只需将文件扩展名从 `.js` 更改为 `.ts`。你还需要更新所有相应的导入语句，将 `import` 语句中的 `.ts` 文件扩展名删除。
+
+> **备注：** 如果在 `import` 语句中使用 `.ts` 文件扩展名，TypeScript 会抛出错误，因此如果你有名为 `./foo.ts` 的文件，你必须将其导入为 `"./foo"`。
+> 更多信息请参阅 TypeScript 手册中的[捆绑程序、TypeScript 运行时和 Node.js 加载器的模块解析](https://www.typescriptlang.org/docs/handbook/2/modules.html##module-resolution-for-bundlers-typescript-runtimes-and-nodejs-loaders)部分。
```

Comment:
I found the url `https://www.typescriptlang.org/docs/handbook/2/modules.html##module-resolution-for-bundlers-typescript-runtimes-and-nodejs-loaders` has been broken (the specified section is not existed), could you update this?

```suggestion
> **备注：** 如果在 `import` 语句中使用 `.ts` 文件扩展名，TypeScript 会抛出错误，因此如果你有名为 `./foo.ts` 的文件，你必须将其导入为 `"./foo"`。更多信息请参阅 TypeScript 手册中的[捆绑程序、TypeScript 运行时和 Node.js 加载器的模块解析](https://www.typescriptlang.org/docs/handbook/2/modules.html##module-resolution-for-bundlers-typescript-runtimes-and-nodejs-loaders)部分。
```